src/lib/hooks/useCurrency.ts

```diff
@@ -134,9 +139,8 @@ export function useTokenFromActiveNetwork(tokenAddress: string | undefined): Tok
   return useMemo(() => {
     // If the token is on another chain, we cannot fetch it on-chain, and it is invalid.
     if (typeof tokenAddress !== 'string' || !isSupportedChain(chainId) || !formattedAddress) return undefined
-
     if (isLoading || !chainId) return null
-    if (!parsedDecimals) return undefined
+    if (parsedDecimals == null) return undefined
```

Comment:
what if it is not null and it is 0 or undefined - in this case shouldn't it be DEFAULT_ERC20_DECIMALS?

Reply:
I don't think it would be safe to default to a potentially invalid decimals value

Reply:
no 0 is a valid decimals case (10^0 = 1, see: https://docs.openzeppelin.com/contracts/3.x/erc20#a-note-on-decimals) 

by using == and not ===, we also capture the case where parsedDecimals is undefined.

Reply:
Btw Jordan on line 126 of useCurrency.ts, when decimals is not available, we are defaulting to DEFAULT_ERC20_DECIMALS which is 18 because ERC20 standard defaults to 18 when the optional extension which allows you to manually set decimals is not used. See documentation here: https://docs.openzeppelin.com/contracts/3.x/erc20#a-note-on-decimals

Reply:
You should not use `==`, because it's called out by linters, and will likely be "fixed" by someone in the future. You should explicitly check for both `null` and `undefined`, at the expense of terseness.

Reply:
ooh my linter doesnt complain when == is used with null, but it does complain with undefined? but i see the point about someone "fixing" it. noted for the future. 

removed line 140 tho due to the other comment about dead code.